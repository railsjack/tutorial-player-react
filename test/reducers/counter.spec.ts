import mainState from '../../app/ui/config/store/reducers/MainState';
import {
  INCREMENT_COUNTER,
  DECREMENT_COUNTER
} from '../../app/ui/config/store/actions/MainState';

describe('reducers', () => {
  describe('mainState', () => {
    it('should handle initial state', () => {
      expect(mainState(undefined, {})).toMatchSnapshot();
    });

    it('should handle INCREMENT_COUNTER', () => {
      expect(mainState(1, { type: INCREMENT_COUNTER })).toMatchSnapshot();
    });

    it('should handle DECREMENT_COUNTER', () => {
      expect(mainState(1, { type: DECREMENT_COUNTER })).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(mainState(1, { type: 'unknown' })).toMatchSnapshot();
    });
  });
});
