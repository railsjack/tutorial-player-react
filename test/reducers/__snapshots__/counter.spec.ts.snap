// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers mainState should handle DECREMENT_COUNTER 1`] = `0`;

exports[`reducers mainState should handle INCREMENT_COUNTER 1`] = `2`;

exports[`reducers mainState should handle initial state 1`] = `0`;

exports[`reducers mainState should handle unknown action type 1`] = `1`;
